src/test/java/org/folio/rest/impl/SendingEmailTest.java

```diff
@@ -234,6 +234,47 @@ public void checkSendingEmailWithDifferentConfigs() throws Exception {
     checkStoredEmailsInDb(emailEntity, DELIVERED);
   }
 
+  @Test
+  public void checkSendingEmailWhileSMPTExceptionHappend() {
+    int maxRecipients = getWiser().getServer().getMaxRecipients();
+    // to get error with status 4xx
+    getWiser().getServer().setMaxRecipients(0);
+    initModConfigStub(userMockServer.port(), getWiserMockConfigurations());
+
+    EmailEntity emailEntity = new EmailEntity()
+      .withNotificationId("1")
+      .withTo(String.format(ADDRESS_TEMPLATE, RandomStringUtils.randomAlphabetic(5)))
+      .withFrom(String.format(ADDRESS_TEMPLATE, RandomStringUtils.randomAlphabetic(7)))
+      .withHeader("Reset password")
+      .withBody("Test text for the message.")
+      .withOutputFormat(MediaType.TEXT_PLAIN);
+
+    Response response = sendEmail(emailEntity)
+      .then()
+      .statusCode(HttpStatus.SC_OK)
+      .extract()
+      .response();
+
+    String expectedMessage = "Error in the 'mod-email' module, the module didn't send email" +
+      " | message: recipient address not accepted: 452 Error: too many recipients";
+    getEmailsShouldBeRetried();
+
+    Response responseDb = getEmails("FAILURE")
+      .then()
+      .statusCode(HttpStatus.SC_OK)
+      .extract()
+      .response();
+
+    EmailEntity result = convertEntriesToJson(responseDb).getEmailEntity().get(0);
+
+    assertEquals(expectedMessage, response.getBody().asString());
+    assertEquals(Integer.valueOf(1), result.getRetryCount());
+    assertEquals(true, result.getShouldRetry());
+
+    // return to initial state
+    getWiser().getServer().setMaxRecipients(maxRecipients);
+  }
+
   @Test
```

Comment:
How about adding test on 500 error?

Reply:
we have a test for failed cause

Reply:
Yes, but we should also add a check that we do not increment the retries in case of 500 error

Reply:
added